docs/source/getting_started.rst

```diff
@@ -0,0 +1,151 @@
+
+- A ``Constraint`` captures a concrete expectation between either two ``DataSource`` s or a single ``DataSource`` and a reference value.
+
+- A ``Requirement`` captures all ``Constraint`` s between two given ``DataSource`` s or all ``Constraint`` s within a single ``DataSource``. If a ``Requirement`` refers links to ``DataSource`` s, it is a ``BetweenRequirement``. If a ``Requirement`` merely refers to a single ``DataSource``, it is a ``WithinRequirement``.
+
+- A specification captures all ``Requirement`` s against a database.
+
+
+Creating a specification
+------------------------
+
+In order to get going, you might want to use the following snippet in a new python file.
+This file will represent a specification.
+
+.. code-block:: python
+
+    import pytest
+    import sqlalchemy as sa
+    from dbcheck.pytest_integration import collect_data_tests
+
+
+    @pytest.fixture(scope="module")
+    def datajudge_engine():
+        # TODO: Adapt connection string to database at hand.
+        return sa.create_engine("your_connection_string")
+
+
+    # TODO: Insert Requirement objects to list.
+    requirements = []
+
+    test_constraint = collect_data_tests(requirements)
+
+This file will eventually lead as an input to pytest. More on that in the section 'Testing a specification'.
+
+
+Specifying Constraints
+----------------------
+
+In order to discover possible ``Constraint`` s, please investigate the ``_add_*_constraint`` methods
+for `BetweenRequirement <https://datajugde.readthedocs.io/en/latest/api/datajudge.requirements.html#datajudge.requirements.BetweenRequirement>`_
+and `WithinRequirement <https://datajugde.readthedocs.io/en/latest/api/datajudge.requirements.html#datajudge.requirements.WithinRequirement>`_
+respectively.
+
+These methods are meant to be mostly self-documenting through usage of expressive parameters.
+
+Note that most ``Constraint`` s will allow for at least one ``Condition``. A ``Condition``
+can be thought of as a conditional event in probability theory or a filter/clause in a datbase
+query. Please consult the doc string of ``Condition`` for greater detail. For examples, please
+see ``tests/unit/test_condition.py``.
+
+Many ``Constraint`` s have optional ``columns`` parameters. If no argument is given, all
+available columns  will be used.
+
+``BetweenRequirement`` s allow for ``Constraint`` s expressing the limitation of a loss or gain (e.g. ``NRowsMinGain``).
+These limitations can often be defined explicitly or be the result of a comparison of date ranges.
+In the latter case the date column must be passed during the instantiation of the ``BetweenRequirement`` and ``date_range_*`` must be passed
+in the respective ``add_*_constraint`` method. When using date ranges as an indicator of change, the ``constant_max_*``
+argument can safely be ignored.
```

Comment:
I feel like a lack context for understanding these two paragraphs. Maybe the text went into details too quickly.